libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -270,93 +581,338 @@ def get_image(layout_object: Any) -> Any:
+                during extraction.
+            extract_tables: Whether to extract tables in a specific format, such as
+                "csv", "markdown", or "html".
+            extract_tables_settings: Optional dictionary of settings for customizing
+                table extraction.
+            **kwargs: Additional keyword arguments for customizing text extraction
+                behavior.
+
+        Returns:
+            This method does not directly return data. Use the `parse` or `lazy_parse`
+            methods to retrieve parsed documents with content and metadata.
+
+        Raises:
+            ValueError: If the mode is not "single" or "page".
+            ValueError: If the extract_tables format is not "markdown", "html",
+            or "csv".
         """
+        super().__init__(extract_images, images_to_text)
+        if mode not in ["single", "page"]:
+            raise ValueError("mode must be single or page")
+        if extract_tables and extract_tables not in ["markdown", "html", "csv"]:
+            raise ValueError("mode must be markdown")
+
+        self.mode = mode
+        self.pages_delimitor = pages_delimitor
+        self.password = password
         self.text_kwargs = text_kwargs or {}
         self.extract_images = extract_images
+        self.extract_tables = extract_tables
+        self.extract_tables_settings = extract_tables_settings
 
     def lazy_parse(self, blob: Blob) -> Iterator[Document]:  # type: ignore[valid-type]
-        """Lazily parse the blob."""
+        """
+        Lazily parse the blob.
+        Insert image, if possible, between two paragraphs.
+        In this way, a paragraph can be continued on the next page.
 
-        import fitz
+        Args:
+            blob: The blob to parse.
 
-        with blob.as_bytes_io() as file_path:  # type: ignore[attr-defined]
-            if blob.data is None:  # type: ignore[attr-defined]
-                doc = fitz.open(file_path)
-            else:
-                doc = fitz.open(stream=file_path, filetype="pdf")
+        Raises:
+            ImportError: If the `pypdf` package is not found.
 
-            yield from [
-                Document(
-                    page_content=self._get_page_content(doc, page, blob),
-                    metadata=self._extract_metadata(doc, page, blob),
+        Yield:
+            An iterator over the parsed documents.
+        """
+        try:
+            import pymupdf
+
+            if not self.extract_tables_settings:
+                from pymupdf.table import (
+                    DEFAULT_JOIN_TOLERANCE,
+                    DEFAULT_MIN_WORDS_HORIZONTAL,
+                    DEFAULT_MIN_WORDS_VERTICAL,
+                    DEFAULT_SNAP_TOLERANCE,
                 )
-                for page in doc
-            ]
 
-    def _get_page_content(self, doc: fitz.Document, page: fitz.Page, blob: Blob) -> str:
+                self.extract_tables_settings = {
```

Comment:
how were these chosen is it possible to add a comment?

Reply:
done